src/const.ts

```diff
@@ -28,3 +28,52 @@ export const enum SensorDeviceClass {
     TEMPERATURE = "temperature",
     VOLTAGE = "voltage"
 }
+
+export const enum UnitOfPower {
+    WATT = "W",
+    KILO_WATT = "kW",
+    BTU_PER_HOUR = "BTU/h",
+}
+
+export const enum UnitOfEnergy {
+    GIGA_JOULE = "GJ",
+    KILO_WATT_HOUR = "kWh",
+    MEGA_JOULE = "MJ",
+    MEGA_WATT_HOUR = "MWh",
+    WATT_HOUR = "Wh",
+}
+
+export const enum UnitOfElectricalCurrent {
+    MILLIAMPERE = "mA",
+    AMPERE = "A"
+}
+
+export const enum UnitOfElectricPotential {
+    MILLIVOLT = "mV",
+    VOLT = "V"
+}
+
+type ConversionRule = {
+    threshold: number;
+    divisor: number;
+    targetUnit: UnitOfEnergy;
+    decimal?: number;
+};
+
+export const unitOfEnergyConversionRules: Record<UnitOfEnergy, ConversionRule[]> = {
```

Comment:
The original function also converted W to kW to MW. This is probably the reason it is not aut scaling all the power values

Reply:
Updated to include power conversion